From: Ubuntu Developers <ubuntu-devel-discuss@lists.ubuntu.com>
Date: Mon, 16 Mar 2026 10:31:00 +0800
Subject: bash-default-editor

# DP: Use `command -v editor`, as an editor, if available.

# DP: Use `command -v editor`, as an editor, if available.
---
 bashline.c      | 4 ++--
 builtins/fc.def | 4 ++--
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/bashline.c b/bashline.c
index 2b4da75..90f075f 100644
--- a/bashline.c
+++ b/bashline.c
@@ -934,8 +934,8 @@ hostnames_matching (const char *text)
    command being entered (if no explicit argument is given), otherwise on
    a command from the history file. */
 
-#define VI_EDIT_COMMAND		"fc -e \"${VISUAL:-${EDITOR:-vi}}\""
-#define EMACS_EDIT_COMMAND	"fc -e \"${VISUAL:-${EDITOR:-emacs}}\""
+#define VI_EDIT_COMMAND		"fc -e \"${VISUAL:-${EDITOR:-$(command -v editor || echo vi)}}\""
+#define EMACS_EDIT_COMMAND	"fc -e \"${VISUAL:-${EDITOR:-$(command -v editor || echo emacs)}}\""
 #define POSIX_VI_EDIT_COMMAND	"fc -e vi"
 
 static int
diff --git a/builtins/fc.def b/builtins/fc.def
index 1d88a49..e53c71a 100644
--- a/builtins/fc.def
+++ b/builtins/fc.def
@@ -181,11 +181,11 @@ uw_unlink (void *fn)
 }
 
 /* String to execute on a file that we want to edit. */
-#define FC_EDIT_COMMAND "${FCEDIT:-${EDITOR:-vi}}"
+#define FC_EDIT_COMMAND "${FCEDIT:-${EDITOR:-$(command -v editor || echo vi)}}"
 #if defined (STRICT_POSIX)
 #  define POSIX_FC_EDIT_COMMAND "${FCEDIT:-ed}"
 #else
-#  define POSIX_FC_EDIT_COMMAND "${FCEDIT:-${EDITOR:-ed}}"
+#  define POSIX_FC_EDIT_COMMAND "${FCEDIT:-${EDITOR:-$(command -v editor || echo ed)}}"
 #endif
 
 int
